testkit/src/lib.rs

```diff
@@ -584,6 +584,11 @@ impl TestKit {
         self.blockchain.snapshot()
     }
 
+    /// Returns the information schema for core.
+    pub fn core_schema(&self) -> CoreSchema<Box<Snapshot>> {
```

Comment:
Nitpick: I don't think the addition of this method is strictly necessary; you could've used

```rust
fn assert_transaction_result<S: AsRef<Snapshot>, T: Transaction>(
    snapshot: S,
    transaction: &T,
    expected_code: u8,
)
```